core/tests/performance_framework/process_data.py

```diff
@@ -0,0 +1,147 @@
+        'loadEventEnd', 'fetchStart', 'domComplete', 'domInteractive',
+        'responseEnd', 'requestStart'
+        ]
+
+    def __init__(self, page_session_stats=None, page_session_timings=None):
+        self.page_session_stats = page_session_stats
+        self.page_session_timings = page_session_timings
+
+        self.page_load_timings = {}
+        if self.page_session_timings:
+            self.page_load_timings = page_session_timings['timing']
+
+        self._validate()
+
+    def _validate(self):
+        """Validates various properties of a PageSessionMetrics object."""
+        if not self.page_session_stats and not self.page_session_timings:
+            raise utils.ValidationError(
+                'Expected atleast one argument among page_session_stats or '
+                'page_session_timings to be present.')
+
+        if self.page_session_stats:
+            if 'log' not in self.page_session_stats:
+                raise utils.ValidationError(
+                    'Expected the page load stats to have a \'log\' entry')
+
+            if 'entries' not in self.page_session_stats['log']:
+                raise utils.ValidationError(
+                    'Expected the log entry of the page load stats to include'
+                    'an additional \'entries\' element')
+
+        if self.page_session_timings:
+            for timing_prop in self.TIMING_PROPERTIES:
+                if timing_prop not in self.page_load_timings:
+                    raise utils.ValidationError(
+                        'Expected the timing entry of the page load timings to'
+                        'include %s property' % timing_prop)
+
+    def get_request_count(self):
+        """Returns the number of requests made prior to the page load
+        completing."""
+        request_count = 0
+
+        request_count = len(self.page_session_stats['log']['entries'])
+
+        if request_count == 0:
+            raise Exception('Total requests cannot be 0.')
+
+        return request_count
+
+    def get_total_page_size_bytes(self):
+        """Returns the total size of a page including all of its resources."""
+        total_size = 0
+
+        for entry in self.page_session_stats['log']['entries']:
+            total_size += int(entry['response']['bodySize'])
+
+        if total_size <= 0:
+            raise Exception('Total page size should be positive.')
+
+        return total_size
+
+    def _get_duration_secs(self, event_end, event_initial):
+        # Check if timestamps are seconds or milliseconds.
+        # From: http://goo.gl/iHNYWx
+        initial_timestamp = self.page_load_timings[event_initial]
+
+        end_timestamp = self.page_load_timings[event_end]
+
+        # If milliseconds convert to seconds.
+        if len(str(initial_timestamp)) >= 13:
+            initial_timestamp /= 1000.0
+
+        if len(str(end_timestamp)) >= 13:
+            end_timestamp /= 1000.0
+
+        duration_secs = end_timestamp - initial_timestamp
```

Comment:
`
duration_sec = (end_timesptamp - initial_timestamp) / 1000.0
`
without converting each of them to sec.